nvtabular/ops/data_schema.py

```diff
@@ -0,0 +1,173 @@
+from .base import Operator
+from .stat_operator import StatOperator
+
+LOG = logging.getLogger("nvtabular")
+
+
+class DataSchema(StatOperator):
+    SCHEMA_FILE_NAME = "schema.pb"
+
+    def __init__(self, tags_by_column, output_path=None):
+        super().__init__()
+        self.schema_path = os.path.join(output_path, self.SCHEMA_FILE_NAME) if output_path else None
+        self.col_names = []
+        self.col_types = []
+        self.col_dtypes = []
+        self.tags_by_column = tags_by_column
+        self.schema = None
+
+    @classmethod
+    def calculate_on_dataset(cls, dataset, tags_by_column, output_path=None, client=None):
+        from nvtabular.column_group import ColumnGroup
+        from nvtabular.workflow import Workflow
+
+        stats = cls(tags_by_column, output_path=output_path)
+
+        col_group = ColumnGroup([])
+
+        for column in dataset.columns:
+            col_group += ColumnGroup(column, tags=tags_by_column.get(column))
+
+        workflow = Workflow(col_group >> stats, client=client)
+        workflow.fit(dataset)
+
+        return stats.schema
+
+    @classmethod
+    def load(cls, directory):
+        from tensorflow_metadata.proto.v0 import schema_pb2
+
+        schema_file = os.path.join(directory, cls.SCHEMA_FILE_NAME)
+
+        LOG.info("Loading schema from %s", schema_file)
+
+        if not os.path.exists(schema_file):
+            return None
+
+        schema = schema_pb2.Schema()
+        with open(schema_file, "rb") as f:
+            schema.ParseFromString(f.read())
+
+        return schema
+
+    def transform(self, columns: Columns, df: DataFrameType) -> DataFrameType:
+        return df
+
+    @annotate("Schema_fit", color="green", domain="nvt_python")
+    def fit(self, columns: Columns, ddf: dd.DataFrame):
+        dask_stats = {}
+
+        ddf_dtypes = ddf.head(1)
+
+        # For each column, calculate the stats
+        for col in columns:
+            dask_stats[col] = {}
+            self.col_names.append(col)
+            # Get dtype for all
+            dtype = ddf_dtypes[col].dtype
+            self.col_dtypes.append(ddf_dtypes[col].dtype)
+
+            domain = ddf[col]
+
+            if str(dtype) == "list":
+                domain = ddf[col].map_partitions(lambda x: x.list.leaves, meta=("x", int))
+                lengths = ddf[col].map_partitions(lambda x: x.list.len(), meta=("x", int))
+                dask_stats[col]["min_length"] = lengths.min()
+                dask_stats[col]["max_length"] = lengths.max()
+                dtype = dtype.leaf_type
+
+            if str(dtype) in ["int8", "int32", "int64", "float32", "float64"]:
+                dask_stats[col]["min"] = domain.min()
+                dask_stats[col]["max"] = domain.max()
+
+        return dask_stats
+
+    def fit_finalize(self, stats):
+        return self.prepare_schema(stats)
+
+    def prepare_schema(self, stats):
+        from tensorflow_metadata.proto.v0 import schema_pb2
```

Comment:
If we are going to depend on this library across frameworks, it seems like we can just import it at the top of the file.